### PR TITLE
Add a security policy

### PR DESCRIPTION
**Pull request by glbrntt**
Motivation:

Security is hugely important to us and our users yet we don't provide
guidelines on how users should report vulnerabilities to us, nor any
commitments we make to resolve these issues.

Modifications:

- Add SECURITY.md detailing how to report vulnerabilities and what
  happens when one is reported.

Result:

It's easier for users to report vulnerabilities to us.

### CODE REVIEW

File: SECURITY.md
Comment (weissi) [inline SECURITY.md:24]: X=3 ?

File: SECURITY.md
Comment (weissi) [inline SECURITY.md:3]: maybe just "security process"? We will issue CVEs but not sure if it's called the CVE process.

File: SECURITY.md
Comment (weissi) [inline SECURITY.md:29]: X=30?

We should add that sometimes it's a coordinated disclosure process (example the HTTP/2 vulns) where the industry coordinates in which case we can't guarantee much.

**Review by weissi — COMMENTED** [review]
Thank you! This looks really good, left a few comments

File: SECURITY.md
Comment (glbrntt) [reply SECURITY.md:24]: Seems totally reasonable.

File: SECURITY.md
Comment (glbrntt) [reply SECURITY.md:29]: Ah, that's a really good point.

**Review by Lukasa — APPROVED** [review]
LGTM as a starting point

**Review by weissi — APPROVED** [review]
Awesome, thank you! Let's get @PeterAdams-A / @tomerd to have a look too

File: SECURITY.md
Comment (PeterAdams-A) [inline SECURITY.md:23]: Should we specify which country?

File: SECURITY.md
Comment (PeterAdams-A) [inline SECURITY.md:11]: I assume you've checked this address is ok for us to use.

Comment (PeterAdams-A) [reply SECURITY.md:11]: Who actually gets those mails?

**Review by PeterAdams-A — COMMENTED** [review]
Generally OK I think.  Added a couple of query comments.

File: SECURITY.md
Comment (glbrntt) [reply SECURITY.md:11]: I assume @tomerd -- but not certain

Comment (tomerd) [reply SECURITY.md:11]: the swift core team has access to this. we are considering creating such email alias for sswg projects which may be applicable here. cc @weissi

File: SECURITY.md
Comment (tomerd) [inline SECURITY.md:26]:
```suggestion
* We will privately inform the Swift Server Work Group ([SSWG][sswg]) of the vulnerability
```

File: SECURITY.md
Comment (tomerd) [inline SECURITY.md:27]: maybe add something like "as required by the sswg security guidelines" and add link to them

**Review by tomerd — APPROVED** [review]
very nice, added couple of comments

File: SECURITY.md
Comment (glbrntt) [reply SECURITY.md:23]: Given emails currently go to core team I switched the country to the US.

File: SECURITY.md
Comment (weissi) [reply SECURITY.md:11]: Yeah, a NIO specific alias that posts to a private forums area would be great actually I think. WDYT other NIO devs?

Comment (Lukasa) [reply SECURITY.md:11]: I think that's a good idea.

Comment (weissi) [reply SECURITY.md:11]: @tomerd is it possible to get a NIO-specific one? @Lukasa / @glbrntt are we waiting for this or can we just merge this PR (I'm in favour of merging now).

Comment (tomerd) [reply SECURITY.md:11]: I reckon we use the sswg-security-reports one and merge. wdyt?

Comment (glbrntt) [reply SECURITY.md:11]: Fine to merge now, we can always update later if/when we get a NIO specific one.

Comment (Lukasa) [reply SECURITY.md:11]: I am somewhat uncomfortable with keeping these two addresses aliased together but that doesn't need to block the merge of this patch.

Comment (tomerd) [reply SECURITY.md:11]: this should be changes to `sswg-security-reports` (SSWG) which has more relevant people on it than `cve@forums.swift.org` (swift core team)

**Pull request merged**
